app-typescript/components/Switch.tsx

```diff
@@ -0,0 +1,48 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface IProps {
+    value?: boolean;
+    disabled?: boolean;
+    onChange(nextValue: boolean): void;
+}
+
+interface IState {
+    checked: boolean;
+}
+
+export class Switch extends React.Component<IProps, IState> {
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
```

Comment:
Remove the internal state and use `this.props.value` instead.

Reply:
If we keep current behaviour for `value` not to be required, then we will need to keep state.